tests/syscall/mount/mount-immutables-unshare-pivot.bats

```diff
@@ -0,0 +1,699 @@
+  [ "$status" -eq 0 ]
+  local inner_pid=$output
+
+  local immutable_mounts=$(list_container_mounts ${syscont} ${inner_pid} "/")
+  run empty_list ${immutable_mounts}
+  [ "$status" -ne 0 ]
+  # Reverse-sort immutable-list to avoid dependency issues (e.g. "/dev/" can't be
+  # unmounted before "/dev/mqueue").
+  immutable_mounts=$(echo ${immutable_mounts} | tr ' ' '\n' | sort -r | tr '\n' ' ')
+
+  for m in ${immutable_mounts}; do
+    # Skip /proc and /sys since these are special mounts (we have dedicated
+    # tests that cover unmounting ops); /dev/null on the other can't be unmounted
+    # as it's being used as entrypoint.
+    if [[ ${m} =~ "/proc" ]] || [[ ${m} =~ "/proc/*" ]] ||
+        [[ ${m} =~ "/sys" ]] || [[ ${m} =~ "/sys/*" ]] ||
+        [[ ${m} == "/dev" ]] || [[ ${m} == "/dev/null" ]]; then
+        continue
+    fi
+
+    printf "\ntesting unmount of immutable mount ${m}\n"
+
+    docker exec ${syscont} sh -c "docker exec inner sh -c \"umount ${m}\""
+    [ "$status" -eq 0 ]
+  done
+
+  docker_stop ${syscont}
+}
+
+# Testcase #2.
+#
+# Ensure that a read-only immutable mount can't be remounted as read-write
+# inside the container.
+@test "immutable ro mount can't be remounted rw -- unshare(mnt) + pivot()" {
+  if [[ $skipTest -eq 1 ]]; then
+    skip
+  fi
+  local_rootfs_prepare
+
+  local syscont=$(docker_run --rm -v ${immutable_ro_dir_path}:${immutable_ro_dir_path}:ro -v ${immutable_ro_file_path}:${immutable_ro_file_path}:ro --mount type=tmpfs,destination=${immutable_masked_dir_path} -v /dev/null:${immutable_masked_file_path} ${CTR_IMG_REPO}/ubuntu-bionic-docker-dbg tail -f /dev/null)
+
+  docker exec -d ${syscont} sh -c "dockerd > /var/log/dockerd.log 2>&1"
+  [ "$status" -eq 0 ]
+
+  wait_for_inner_dockerd ${syscont}
+
+  docker exec ${syscont} sh -c "docker run --privileged -d --name inner -v ${immutable_ro_dir_path}:${immutable_ro_dir_path}:ro -v ${immutable_ro_file_path}:${immutable_ro_file_path}:ro --mount type=tmpfs,destination=${immutable_masked_dir_path} -v /dev/null:${immutable_masked_file_path} ${CTR_IMG_REPO}/ubuntu:latest tail -f /dev/null"
+  [ "$status" -eq 0 ]
+  docker exec ${syscont} sh -c "docker inspect --format='{{.State.Pid}}' inner"
+  [ "$status" -eq 0 ]
+  local inner_pid=$output
+
+  local immutable_ro_mounts=$(list_container_ro_mounts ${syscont} ${inner_pid} "/")
+  run empty_list ${immutable_ro_mounts}
+  [ "$status" -ne 0 ]
+
+  for m in ${immutable_ro_mounts}; do
+    printf "\ntesting rw remount of immutable ro mount ${m}\n"
+
+    docker exec ${syscont} sh -c "docker exec inner sh -c \"mount -o remount,bind,rw ${m}\""
+    [ "$status" -ne 0 ]
+  done
+
+  local immutable_ro_mounts_after=$(list_container_ro_mounts ${syscont} ${inner_pid} "/")
+  [[ $immutable_ro_mounts == $immutable_ro_mounts_after ]]
+
+  docker_stop ${syscont}
+}
+
+# Testcase #3.
+#
+# Ensure that a read-write immutable mount *can* be remounted as read-only inside
```

Comment:
Same as before: this test applies to bind-mounts from immutable mounts, not to the original immutable mounts themselves.